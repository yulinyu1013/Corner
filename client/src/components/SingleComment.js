import React from 'react';
import logo from '../images/logo.png';
import '../styles/singlecomment.css';
import { deletePost } from '../fetchers/feedsApi';

function SingleComment({
  singlecomment,
  setIsEditing,
  setCommentText,
  allComments,
  setAllComments,
  userId,
}) {
  const dateTime = new Date(singlecomment.updatedAt).toLocaleDateString();

  const onEditing = () => {
    setIsEditing({ condition: true, id: singlecomment.id });
    setCommentText(singlecomment.text);
  };

  const onDelete = () => {
    deletePost(singlecomment.id).then(() => {
      const updated = allComments.filter((post) => post.id !== singlecomment.id);
      setAllComments(updated);
    });
  };

  return (
    <div className="single-comment">
      <div className="top-content">
        <img className="comment-pic" src={singlecomment.avatar ? singlecomment.avatar : logo} alt="commentUser" />
        <div className="grey-box">
          <p>{singlecomment.creatorName}</p>
          <p>{singlecomment.text}</p>
        </div>
      </div>
      <div className="comment-operation">
        {singlecomment.creator === userId && <p className="icon" onClick={onEditing} role="presentation">edit</p>}
        {singlecomment.creator === userId && <p className="icon" onClick={onDelete} role="presentation">delete</p>}
        <p>{dateTime}</p>
      </div>
    </div>
  );
}

export default SingleComment;
