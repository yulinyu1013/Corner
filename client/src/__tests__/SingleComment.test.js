import React from 'react';
import renderer from 'react-test-renderer';
import SingleComment from '../components/SingleComment';

const comment =   {
  id: 3,
  creator: 1,
  creatorName: 'Jhon',
  corner: 'test_corner',
  cornerName: 'test_corner',
  type: 'comment',
  pid: 1,
  text: 'first test comment',
  pic: [],
  audio: null,
  video: null,
  createdAt: '2021-08-16T23:00:33.010+02:00',
  updatedAt: '2021-08-16T23:00:33.010+02:00',
  isFlagged: 0,
  isHidden: 0,
}

it('renders SingleComment correctly', () => {
  const tree = renderer.create(<SingleComment singlecomment={comment} userId='5' />).toJSON();
  expect(tree).toMatchSnapshot();
});