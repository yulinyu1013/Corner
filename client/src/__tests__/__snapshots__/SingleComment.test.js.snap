// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SingleComment correctly 1`] = `
<div
  className="single-comment"
>
  <div
    className="top-content"
  >
    <img
      alt="commentUser"
      className="comment-pic"
      src="logo.png"
    />
    <div
      className="grey-box"
    >
      <p>
        Jhon
      </p>
      <p>
        first test comment
      </p>
    </div>
  </div>
  <div
    className="comment-operation"
  >
    <p>
      8/16/2021
    </p>
  </div>
</div>
`;
